database/redis/notification.go

```diff
@@ -98,7 +130,119 @@ func (connector *DbConnector) removeNotifications(notifications []*moira.Schedul
 }
 
 // FetchNotifications fetch notifications by given timestamp and delete it
-func (connector *DbConnector) FetchNotifications(to int64) ([]*moira.ScheduledNotification, error) {
+func (connector *DbConnector) FetchNotifications(to int64, limit int64) ([]*moira.ScheduledNotification, error) {
+	// No limit
+	if limit == 0 {
+		return connector.fetchNotificationsNoLimit(to)
+	}
+
+	count, err := connector.notificationsCount(to)
+	if err != nil {
+		return nil, err
+	}
+
+	// Hope count will be not greater then limit when we call fetchNotificationsNoLimit
+	if count == nil {
+		return nil, fmt.Errorf("failed to count notification in db")
+	}
+	if *count < limit/2 {
```

Comment:
I think that this is heuristic and this heuristic can work in wrong way. Imagine that limit is set to 2. In this case one new notification will break this heuristic.

Reply:
Well, let's apply this heuristic only if limit large enough.

I know it may return a little bit more notifications than limit, but still I would like to use the heuristic because I don't want to run all the "WATCH" machinery once again.